src/main/kotlin/com/ampnet/reportserviceeth/controller/ReportingController.kt

```diff
@@ -49,4 +56,11 @@ class ReportingController(private val reportingService: ReportingService) {
         val pdfContents = reportingService.generatePdfReportForUserTransaction(transactionServiceRequest)
         return ResponseEntity(pdfContents, ControllerUtils.getHttpHeadersForPdf(), HttpStatus.OK)
     }
+
+    fun getReportPeriodForAddress(from: LocalDate?, to: LocalDate?, address: String) =
+        PeriodServiceRequest(
+            from?.let { LocalDateTime.of(from, LocalTime.MIN) }
+                ?: userService.getUser(address.lowercase()).createdAt.toLocalDateTime(),
```

Comment:
Why fetch the same user two times? Frist time here, the second time in `TemplateDataServiceImpl`

Reply:
Yeah I just realised that the period selection is already implemented.